### PR TITLE
Popover changes

### PR DESCRIPTION
**Pull request by davidtheclark**
There are a lot of changes in this PR that are mostly just reformatting comments or improving documentation and, especially, documentation examples. But there are also some very important changes, which I hope are improvements, to Popover, PopoverTrigger, and Tooltip.

- There are 3 standard popover styles: light, dark, and warning. 
- `theme*` props have been replaced with higher-level styling props like `coloring` (light, dark, or warning) and `padded` (true or false).
- A new `passthroughProps` convention is introduced to pass props through a component to some child (most often to pass it through directly to a DOM element).
- Children of Tooltip no longer need to be functions!

### CODE REVIEW

**Comment by davidtheclark** [comment]
@jseppi could you give this a review? There's a lot I think you could skip over, but if the changes and documentation make sense to you, and the examples (in docs and test cases) are clear and function as you'd expect, I think we'll be good to go.

**Comment by jseppi** [comment]
👁‍🗨 Checking it out...

File: src/components/popover-trigger/popover-trigger.js [hunk 1]
```diff
@@ -414,3 +383,93 @@ export default class PopoverTrigger extends React.Component {
...
+PopoverTrigger.propTypes = {
+  /**
+   * The trigger content. This can either be a string or valid JSX.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Accepts a CSS `display` value for the trigger: `'inline-block'` or
+   * `'block'`.
+   */
+  display: PropTypes.string,
```
Comment (jseppi) [inline src/components/popover-trigger/popover-trigger.js]: Maybe use `PropTypes.oneOf(['inline-block', 'block'])`

File: src/components/popover-trigger/popover-trigger.js [hunk 2]
```diff
@@ -414,3 +383,93 @@ export default class PopoverTrigger extends React.Component {
...
+  respondsToFocus: PropTypes.bool,
+  /**
+   * Whether or not the trigger is disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * Props to pass directly to the [Popover](#popover) component. See Popover's
+   * documentation for details.
+   */
+  popoverProps: PropTypes.object,
```
Comment (jseppi) [inline src/components/popover-trigger/popover-trigger.js]: Maybe export Popover's propTypes and use it here?

File: src/components/popover-trigger/popover-trigger.js (hunk 1 above)
Comment (davidtheclark) [reply src/components/popover-trigger/popover-trigger.js]: Ah, yeah, intended to do that.

File: src/components/popover-trigger/popover-trigger.js (hunk 2 above)
Comment (davidtheclark) [reply src/components/popover-trigger/popover-trigger.js]: I worry that that might make it harder for babel-plugin-remove-prop-types to remove them, and I think the runtime check will work either way.

**Pull request merged**